File: 01-Two-Sum/solution.py
# Two Sum Problem

nums = [2, 7, 11, 15]
target = 9

# Dictionary to store number and its index
mp = {}

for i in range(len(nums)):

    # Find the required complement
    need = target - nums[i]

    # Check if complement already exists
    if need in mp:
        print("Indices:", mp[need], i)
        break

    # Store current number and its index
    mp[nums[i]] = i

File: 01-Two-Sum/solution.c
#include <stdio.h>

int main() {

    int nums[] = {2,7,11,15};

    int target = 9;

    int n = sizeof(nums)/sizeof(nums[0]);

    for(int i=0;i<n;i++){

        for(int j=i+1;j<n;j++){

            if(nums[i]+nums[j]==target){

                printf("Indices: %d %d",i,j);

                return 0;

            }

        }

    }

}